### ISSUE TITLE
When the @wordpress/edit-site package is imported, multiple errors occur in the post editor (navigation block fails to render)

### ISSUE THREAD

**Issue by briangrider** [WordPress/gutenberg#54252]
### Description

When both the @wordpress/edit-site and @wordpress/edit-post packages are imported, an error is thrown from the registry and navigation menus break in the post editor.  

The first error that is thrown in the console is: `Store "core/interface" is already registered.` at line 220 in registry.js. I've encountered this error in the console for a long time now but I thought this wasn't an issue and continued with development. I'm now seeing the issue manifest where navigation menus in the post editor fail to render and instead show: `This block has encountered an error and cannot be previewed.` This is accompanied by the console error: 'TypeError: Cannot destructure property 'params' of 'Mt(...)' as it is undefined.' on line 23 of navigation-menu-edit.js.

I've tried conditionally rendering the components as well as even using React's 'lazy' with Suspense but the issue persists.

For testing purposes, I am importing the PluginMoreMenu from both the @wordpress/edit-site package and the @wordpress/edit-post package but I believe I can also reproduce the problem just by importing any component from '@wordpress/edit-site' (tested with PluginMoreMenu and PluginSidebar) and opening the post-editor. The `Store "core/interface" is already registered.` error is still thrown and all navigation menus on the post fail to load with the same error listed above.

### Step-by-step reproduction instructions

1. Create a custom Wordpress plugin
2. Using npm, import components from the '@wordpress/edit-site' and '@wordpress/edit-post' packages
3. You can choose to render, not render, or conditionally render these components but the issue should occur in the post editor whether those components are rendered or not. It is the mere existence of the import from the '@wordpress/edit-site' package that causes the error in the post editor
4. Import the compiled js file with the enqueue_block_editor_assets action in the plugin
5. Add a navigation block to a new post on the post editor - this should fail to render with the console error: 'TypeError: Cannot destructure property 'params' of 'Mt(...)' as it is undefined.' on line 23 of navigation-menu-edit.js. The 'Store "core/interface" is already registered.' error will be logged to the console immediately on page load.
6. Remove the '@wordpress/edit-site' import, rebuild the export, and reload the post on the post editor. No errors are thrown and the navigation menu now renders as expected

Similarly, the core/interface error is still logged to the console when using the site editor while the @wordpress/edit-post package is imported, but navigation menus successfully render in this context

### Screenshots, screen recording, code snippet

import { PluginMoreMenuItem } from '@wordpress/edit-post';
import { PluginMoreMenuItem as PluginMoreMenuItemSite } from '@wordpress/edit-site;

### Environment info

WP 6.3.1, Gutenberg 16.5.1, Twenty twenty-three v. 1.2

Chrome on a Windows 10 desktop

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by briangrider** [comment]
I've bundled up a plugin to easily test this:

All it does is import a js file that has this code:

```
import { PluginMoreMenuItem } from '@wordpress/edit-post';
import { PluginMoreMenuItem as PluginMoreMenuItemSite } from '@wordpress/edit-site';
import { registerPlugin } from '@wordpress/plugins';

registerPlugin( 'edit-site-test', {
	render: () => <></>,
} );
```

On a fresh install of WP 6.3.1 without any other plugins (including gutenberg) the problem is reproducible with just this code. I included the package.json and src folder as well.

Here is the plugin:
[test-edit-site.zip](https://github.com/WordPress/gutenberg/files/12552477/test-edit-site.zip)

**Comment by tresorama** [comment]
I've expeienced the same but i didn't know what it was derived from.
Block core/navigation broken included.

**Comment by briangrider** [comment]
> I've expeienced the same but i didn't know what it was derived from. Block core/navigation broken included.

Yeah, it's easy enough to reproduce the problems so hopefully it's an easy fix.

**Comment by LarsEjaas** [comment]
I also experience this issue,  importing from the @wordpress/edit-post package

**Comment by ndiego** [comment]
Yes, I can confirm this as well. If you import something from `@wordpress/edit-post`, you will get the error in the Site Editor. If you import something from `@wordpress/edit-site`, you will get the error in the Post Editor. 

@youknowriad I believe this issue will naturally resolve itself as we continue to unify both editors and once things are moved to the `@wordpress/editor,` right? https://github.com/WordPress/gutenberg/pull/60033 is a good reference.

**Comment by youknowriad** [comment]
edit-post and edit-site are mutually exclusive. They are not meant to be imported in one another as they represent the top level scripts for either the post editor or site editor pages.

Longer term, there will be only one of these that will remain probably. But for now, I think for me this is "doing it wrong". If there's something you need from one package that is not present on the other or something like that, let's consider opening a specific issue for this use-case to address it.

**Comment by briangrider** [comment]
> edit-post and edit-site are mutually exclusive. They are not meant to be imported in one another as they represent the top level scripts for either the post editor or site editor pages.
> 
> Longer term, there will be only one of these that will remain probably. But for now, I think for me this is "doing it wrong". If there's something you need from one package that is not present on the other or something like that, let's consider opening a specific issue for this use-case to address it.

I think the biggest issue here is that if you are importing a slotfill for when the user is in the site editor, and in the same plugin, you are importing a slotfill for when the user is in the post editor, there is no way to conditionally import from edit-post ONLY when in the post editor and import from edit-site ONLY when in the site editor (at least no way I could figure out that got rid of that error). The error pops up no matter what, even if the slotfill is conditionally used. 

So currently, your options are to:
a. only use the slotfills from edit-post
b.  only use the slotfills from edit-site
or c. always have a JS error

In the plugin I've been working on, I just decided not to use any slotfills from edit-post or edit-site but it was an unfortunate decision that felt more like "well this is broken so I guess I won't use it" rather than "I don't need this so I won't use it."

**Comment by youknowriad** [comment]
@briangrider Did you consider using two separate scripts for your plugin, one for editPost and one for editSite? In the next releases, we'll be moving most (all maybe) of the slots to the "editor" package which means you won't have to use neither editPost or editSite directly.

**Comment by briangrider** [comment]
> @briangrider Did you consider using two separate scripts for your plugin, one for editPost and one for editSite? In the next releases, we'll be moving most (all maybe) of the slots to the "editor" package which means you won't have to use neither editPost or editSite directly.

Glad to hear the slots will be unified in the editor package. I did try various types of workarounds (including splitting scripts) but it just didn't make sense. There was always some amount of code duplication, or conceptually odd programming patterns for what seemed like it really should just be a single import that works without an error. So I ended up dropping the imports altogether but glad I can bring those features back with the next releases.
